lib/types.d.ts

```diff
@@ -0,0 +1,1766 @@
+export function typeParameterInstantiation(params: Array<Flow>): TypeParameterInstantiation;
+export function unionTypeAnnotation(types: any): UnionTypeAnnotation;
+export function voidTypeAnnotation(): VoidTypeAnnotation;
+export function jSXAttribute(name: JSXIdentifier | JSXNamespacedName, value?: JSXElement | JSXFragment | StringLiteral | JSXExpressionContainer): JSXAttribute;
+export function jSXClosingElement(name: JSXIdentifier | JSXMemberExpression): JSXClosingElement;
+export function jSXElement(openingElement: JSXOpeningElement, closingElement?: JSXClosingElement, children: Array<JSXText | JSXExpressionContainer | JSXSpreadChild | JSXElement | JSXFragment>, selfClosing: any): JSXElement;
+export function jSXEmptyExpression(): JSXEmptyExpression;
+export function jSXExpressionContainer(expression: Expression): JSXExpressionContainer;
+export function jSXSpreadChild(expression: Expression): JSXSpreadChild;
+export function jSXIdentifier(name: string): JSXIdentifier;
+export function jSXMemberExpression(object: JSXMemberExpression | JSXIdentifier, property: JSXIdentifier): JSXMemberExpression;
+export function jSXNamespacedName(namespace: JSXIdentifier, name: JSXIdentifier): JSXNamespacedName;
+export function jSXOpeningElement(name: JSXIdentifier | JSXMemberExpression, attributes: Array<JSXAttribute | JSXSpreadAttribute>, selfClosing?: boolean): JSXOpeningElement;
+export function jSXSpreadAttribute(argument: Expression): JSXSpreadAttribute;
+export function jSXText(value: string): JSXText;
+export function jSXFragment(openingFragment: JSXOpeningFragment, closingFragment: JSXClosingFragment, children: Array<JSXText | JSXExpressionContainer | JSXSpreadChild | JSXElement | JSXFragment>): JSXFragment;
+export function jSXOpeningFragment(): JSXOpeningFragment;
+export function jSXClosingFragment(): JSXClosingFragment;
+export function noop(): Noop;
+export function parenthesizedExpression(expression: Expression): ParenthesizedExpression;
+export function awaitExpression(argument: Expression): AwaitExpression;
+export function bindExpression(object: any, callee: any): BindExpression;
+export function classProperty(key: any, value?: Expression, typeAnnotation?: TypeAnnotation | TSTypeAnnotation | Noop, decorators?: Array<Decorator>, computed?: boolean, abstract?: boolean, accessibility?: "public" | "private" | "protected", optional?: boolean, readonly?: boolean, _static?: boolean): ClassProperty;
+export function decorator(expression: Expression): Decorator;
+export function doExpression(body: BlockStatement): DoExpression;
+export function exportDefaultSpecifier(exported: Identifier): ExportDefaultSpecifier;
+export function exportNamespaceSpecifier(exported: Identifier): ExportNamespaceSpecifier;
+export function tSParameterProperty(parameter: Identifier | AssignmentPattern, accessibility?: "public" | "private" | "protected", readonly?: boolean): TSParameterProperty;
```

Comment:
Ugh... Can you generate these names like I did in https://github.com/babel/babel/pull/6967?

Reply:
I agree that the `tS` prefix is ugly, and would also prefer `ts`. The downside is that would not be backwards-compatible (see [previous discussion](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/21972#issuecomment-351729657)).

As noted in that discussion, a good third option is a `TS` prefix - it's both backwards-compatible and consistent, and as an added benefit it shares its type's name. A downside is that Flow uses a single namespace for types and values, and will not let us export both a function and a type with, say, the name `TSParameterProperty`. Here's my suggestion:

1. For now, let's rename all of these to `TS...` in just the TS typings. For example, `tSParameterProperty` would become `TSParameterProperty`
2. When we're ok shipping a breaking version, let's rename all of them to `ts...` in both the TS and Flow typings. For example, `tSParameterProperty` would become `tsParameterProperty`

Of course, if we're shipping this as part of v7 (which is breaking anyway), let's do (2) now and skip (1).

Reply:
Since we are still in beta, we can do (2).

Reply:
Done for TS and Flow, and updated docs.